src/Authenticator/SessionAuthenticator.php

```diff
@@ -57,10 +58,11 @@ public function authenticate(ServerRequestInterface $request, ResponseInterface
         }
 
         if ($this->getConfig('identify') === true) {
-            $user = $this->identifiers()->identify([
-                'username' => $user[$this->getConfig('fields')['username']],
-                'password' => $user[$this->getConfig('fields')['password']]
-            ]);
+            $credentials = [];
+            foreach ($this->getConfig('fields') as $key => $field) {
+                $credentials[$key] = $user[$field];
+            }
+            $user = $this->identifiers()->identify($credentials);
```

Comment:
Identifying users from the session on username only looks good now that FormAuthenticator whitelists its fields :+1: